app/models/compute_resource.rb

```diff
@@ -77,6 +77,10 @@ def self.providers_requiring_url
     _("Libvirt, oVirt, OpenStack and Rackspace")
   end
 
+  def self.providers_url_mandatory
+    %w(libvirt ovirt openstack rackspace)
```

Comment:
Can we make this an option of the Model-class, e.g. `Foreman::Model::Ovirt`? It would be easier for plugins to change this.